v3/mathjax3-ts/util/Options.ts

```diff
@@ -0,0 +1,218 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2017 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements functions for handling option lists
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+
+/*****************************************************************/
+/*
+ *  Check if an object is an object literal (as opposed to an instance of a class)
+ */
+
+const OBJECT = {}.constructor;
+function isObject(obj: any) {
+    return typeof obj === 'object' && obj !== null && obj.constructor === OBJECT;
+}
+
+/*****************************************************************/
+/*
+ * Generic list of options
+ */
+export type OptionList = {[name: string]: any};
+
+/*****************************************************************/
+/*
+ *  Used to append an array to an array in default options
+ *  E.g., an option of the form
+ *
+ *    {
+ *      name: {APPEND: [1, 2, 3]}
+ *    }
+ *
+ *  where 'name' is an array in the default options would end up with name having its
+ *  original value with 1, 2, and 3 appended.
+ */
+export const APPEND = Symbol('Append to option array');
+
+
+/*****************************************************************/
+/*
+ * Get all keys and symbols from an object
+ *
+ * @param{Optionlist} def        The object whose keys are to be returned
+ * @return{(string | symbol)[]}  The list of keys for the object
```

Comment:
`symbol` is the primitive type (the actual symbol), whereas `Symbol` is the constructor for a `symbol`.